### DISCUSSION TITLE
check version of tycho-build equals version of tycho.core?

### DISCUSSION THREAD

**Discussion by Bananeweizen** [eclipse-tycho/tycho#4531] [Ideas]
Most developers use a shared property for the version of all tycho maven plugins. Better developers additionally use maven-enforcer with the requireSameVersions rule to enforce all tycho maven plugins are in fact the same version.

However, the tycho-build extension in `extensions.xml` cannot be held in sync with either of those 2 approaches. I've seen multiple times that the POM refers to the most recent Tycho, but the extensions.xml is outdated. Would it make sense to have a runtime check that verifies these 2 are equal? It's probably okay most of the time to have 2 different versions, but especially with major version increases this seems like a potential source of bugs due to incompatible changes.

Inspired by https://github.com/eclipse-equinox/equinox/pull/724

**Comment by laeubi** [comment]
This one shows how to use one shared version with extension + build plugins
https://tycho.eclipseprojects.io/doc/latest/StructuredBuild.html

Often it is good to have them equal but there are many cases where a (slight) deviation works without a problem, this is mostly due how maven-extensions work and how tycho uses them.

**Comment by Bananeweizen** [comment]
Ah, you move the property out to the shared maven config, nice idea. I'll have to remember that for projects where dependabot is not enabled.

**Reply by laeubi** [reply]
I don't know if this is supported by dependabot but I use that technique whenever possible as it is not so hard to update the version manually :-)
